internal/host/hostcommands/install_cmd.go

```diff
@@ -114,6 +114,11 @@ func (i *installCmd) getFullInstallerCommand(cluster *common.Cluster, host *mode
 		BootDevice:           swag.String(bootdevice),
 	}
 
+	// The HA mode is not relevant for workers.
+	if string(role) == models.HostUpdateParamsHostRoleWorker {
```

Comment:
Is this change required ? how did it work before ?

Reply:
The assisted-installer's CLI param validation checks that if the HA mode is set to "None" then the role has to be master/bootstrap, which (currently) makes sense in day-1 - HA mode means SNO, SNO means there's just one node, and that node has to be master/bootstrap.

But for workers, in general, it doesn't matter if they're day-1 or day-2 - the HA mode doesn't make any difference. Workers don't care what kind of cluster they're going to join. 

How it worked so far -
- In ZTP, we have the HA mode.
- In REST, we don't have the HA mode (it's empty).

You can see on line 100 when `cluster.HighAvailabilityMode` is `nil` we just default to assuming it's `models.ClusterHighAvailabilityModeFull`, which caused the CLI param validation I mentioned in the beginning to be skipped.

But in ZTP, and in SNO, that validation started failing, because we're trying to get a worker to join a non-HA cluster.

@paul-maidment made a similar change on the installer and agent to accept a `nil` HA mode - this is useful for workers where we don't care about the HA. These changes would have to be merged before this PR does